### ISSUE TITLE
Allow to not collect personal data

### ISSUE THREAD

**Issue by olafgrabienski** [backdrop-contrib/statistics#3]
The module stores data about "the remote host's IP address" (see README) and other things. It would be good to check which kind of stored data is "personal data" according to the General Data Protection Regulation of the European Union (see https://en.wikipedia.org/wiki/General_Data_Protection_Regulation) and to allow to not collect / store respective data.

(More a general suggestion than a request, as I'm not using this module.)

**Comment by docwilmot** [comment]
I dont think this is feasible. It would be difficult to have a module collecting statistics which did not collect such data. I think it would be up to the site owner to have a GDPR opt-in module to address this.

**Comment by olafgrabienski** [comment]
> It would be difficult to have a module collecting statistics which did not collect such data.

Sure, some data should be collected but not all of the data is necessarily important. Take the IP address as an example: While the IP is considered as personal data according to the GDPR it's not that useful for each analytics use case. An option to disable (or anonymize) it would be great, in my opinion. If I'm primarily interested in page view quantities, the *Statistics* module would continue to be useful.

I still don't use the module for the time being. I'm however interested to find out which kind of collected data is "personal" according to the GDPR, hoping to identify a subset of data which could be disabled.

**Comment by albanycomputers** [comment]
One possible option...

Add a new field "blocked" to the DB... Hash the IP address with the system-wide unique site private key to use as a salt... this creates a unique hash...

If later the admin wants to ban a bad actor's activity, they flag it for banning...

Logic flow...

A visitor comes to the site, and we hash their IP. If it's in the DB as blocked, we store the actual IP, not the hash, and then send the IP to the blocking/banning software, if installed (and/or update the blocking/banning module).

There's a bit more logic involved, but this is the gist of it. We could update the banning/blocking module to also use the hashing mechanism. 

Input, critique, and thoughts welcomed.
